http/echo_server.py

```diff
@@ -0,0 +1,99 @@
+#! /usr/bin/env python
+
+import socket
+import email.utils
+import os
+import sys
+import mimetypes
+
+
+def echo_servers():
+    """Initiates a socket from the server side"""
+    server_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(1)
+
+    try:
+        while True:
+            done = False
+            text_received = ''
+            while not done:
+                conn, addr = server_socket.accept()
+                response = conn.recv(32)
+                if len(response) < 32:
+                    done = True
+                text_received = "{}{}".format(text_received, response)
+            if text_received:
+                response_user = parase_request(text_received)
+            conn.sendall(response_user)
+    except KeyboardInterrupt:
+       server_socket.close()
+
+def response_ok(content, body_text):
+    first_line = 'HTTP/1.1 200 OK'
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header = 'Content-Type: {}'.format(content)
+    body = body_text
+    size = str(sys.getsizeof(body))
+    end = '\r\n'
+    response = [first_line, timestamp, header, size, body, end]
+    return end.join(response)
+
+def response_error(error_code, error_message):
+    first_line = 'HTTP/1.1 {} {}'.format(error_code, error_message)
+    timestamp = 'Date: ' + email.utils.formatdate(usegmt=True)
+    header = 'Content-Type: text/plain'
+    body = '{} {}'.format(error_code, error_message)
+    size = str(sys.getsizeof(body))
+    end = '\r\n'
+    error_response = [first_line, timestamp, header, size, body, end]
+    return end.join(error_response)
+
+def parase_request(client_request):
+    full_text = client_request.splitlines()
+    first_line = full_text[0].split(' ')
```

Comment:
If you really mean "split on whitespace", then you really don't need to provide an argument to split.  In fact, it's actually problematic to do so.  What happens if there are *extra* spaces between words in the string you're splitting?

```python
>>> foo = "text with  extra   spaces"
>>> foo.split(" ")
['text', 'with', '', 'extra', '', '', 'spaces']
>>> foo.split()
['text', 'with', 'extra', 'spaces']
```